### PR TITLE
Add the webmidi IDL file + test

### PR DESCRIPTION
**Pull request by lukebjerring**
Hello reviewer(s),

This PR is intended to consolidate the spec’s IDL definition with the WPT test suite’s copy, and any idlharness tests.

The up-to-date copy of the IDL file was automatically extracted from the reffy-reports repo (https://github.com/tidoust/reffy-reports/tree/master/whatwg/idl) which scrapes known specs automatically + regulary.

This PR is part of a migration project which will eventually be automatically updating and creating PRs for changes in spec IDL.

Please check that:
The Spec (and its source) is correct and up-to-date
All tests which cover the IDL in the spec have been migrated to fetch + use the idl in the `interfaces/` directory (instead of inline copies in the test files).

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build BROKEN

Started: 2018-03-05 22:57:18
Finished: 2018-03-05 23:01:48


<h2>Failing Jobs</h2>
<ul>

<li>firefox:nightly</li>

<li>chrome:dev</li>

<li>safari:11.0</li>

<li>MicrosoftEdge:16.16299</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/25302)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/349446134)

**Comment by lukebjerring** [comment]
@cwilso - can you take a look?

**Comment by lukebjerring** [comment]
@jussi-kalliokoski are you able to take a look?

File: webmidi/META.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+spec: https://webaudio.github.io/web-midi-api/
+suggested_reviewers:
+  - cwilso
```
Comment (foolip) [inline webmidi/META.yml]: @cwilso @jussi-kalliokoski, I've sent org invites which you can accept at https://github.com/web-platform-tests. If you don't want to review tests for Web MIDI in this repo, can you suggest some good candidates?

**Review by foolip — APPROVED** [review]

File: webmidi/META.yml (hunk 1 above)
Comment (cwilso) [reply webmidi/META.yml]: I'll review.  Sorry, been kinda backed up.

**Review by cwilso — APPROVED** [review]
LGTM.

**Comment by lukebjerring** [comment]
@foolip - rejigged to use `idl_test`. PTAL. I also threw the `input` and `output` objects behind an if statement, since we make no effort to automate these (yet).

@cwilso is there any existing automation for testing around, say, virtual/software devices?

**Review by foolip — APPROVED** [review]
Tested manually, looks good. Only failure in Chrome is because of this TODO: https://cs.chromium.org/chromium/src/third_party/blink/renderer/modules/webmidi/midi_output.idl?l=34&rcl=378384acf5e76872e88c144e4ab8d154ce533f84

**Pull request merged**